<?xml version="1.0" encoding="utf-8"?>

<launch>
  <arg name="distobox_usb_port"
       default="/dev/serial/by-id/usb-Silicon_Labs_CP2103_USB_to_UART_Bridge_Controller_0001-if00-port0"
       doc="Path of the USB port that the distobox is connected at."/>
  <!-- pointlaser_comm node -->
  <node name="pointlaser_comm"
        pkg="cpt_pointlaser_comm"
        type="cpt_pointlaser_comm"
        output="screen"
        respawn="true">
    <param name="port" value="$(arg distobox_usb_port)"/>
  </node>

  <!-- First Laser (A): Finger pointing towards endeffector.
       Should be connected to topmost connector board. -->
  <node pkg="tf" type="static_transform_publisher"
        name="pointlaser_calibration_a"
        args="0.0082 -0.0353 -0.1446 0 0.707 0 0.707  /kinova_link_6 /pointlaser_A 100"/>
  <!-- Second Laser: Finger pointing towards distobox.
       Should be connected to middle connector board. -->
  <node pkg="tf" type="static_transform_publisher"
        name="pointlaser_calibration_b"
        args="0.023 0.0353 -0.1402 0 0 0 1  /kinova_link_6 /pointlaser_B 100"/>
  <!-- Third Laser: "The Crown", not one of the other two.
       Should be connected to down connector board. -->
  <node pkg="tf" type="static_transform_publisher"
        name="pointlaser_calibration_c"
        args="0.0353 -0.018 -0.1402 0  0 -0.707 0.707  /kinova_link_6 /pointlaser_C 100"/>
</launch>
